### PR TITLE
TextInputs announce errors and set aria flags

### PR DESCRIPTION
**Pull request by jish**
**Jira:**

[[IL-1287] Make TextInput error messages screen readable](https://clever.atlassian.net/browse/IL-1287)

**Overview:**

Aria attributes can be set on the client side or on the server side.
If they are set by the client, we need to notify screen readers that
something has been changed and needs to be announced. So we need an
`aria-live` tag on first render. If the contents of this `aria-live`
element change, the screen reader will be notified.

**Screenshots/GIFs:**

**Testing:**
- [ ] Unit tests
- Manual tests:
  - [ ] Chrome
  - [ ] Safari
  - [ ] IE10

**Roll Out:**
- Before merging:
  - [ ] Updated docs
  - [ ] Bumped version in `package.json`
    - Breaking change? Run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)

### CODE REVIEW

File: test/TextInput_test.tsx [hunk 1]
```diff
@@ -63,9 +63,11 @@ describe("TextInput", () => {
...
 
     // check the span element with error message
-    assert.equal(textInput.find("span").length, 1);
-    assert.equal(textInput.find("span").text(), "<FontAwesome /> This is an error");
-    assert(textInput.find("span").hasClass("TextInput--error"));
+    assert.equal(textInput.find("span.TextInput--error").length, 1);
+    assert.equal(
+      textInput.find('span[aria-live="polite"]').text(),
+      "<FontAwesome /> This is an error",
+    );
```
Comment (lubalexuan) [inline test/TextInput_test.tsx]: 👍

File: src/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -179,6 +180,7 @@ export class TextArea extends React.Component<Props, State> {
     const inputNote = this.renderNote(errorMessage);
 
     const textAreaProps = {
+      ["aria-invalid"]: !valid,
```
Comment (lubalexuan) [inline src/TextArea/TextArea.tsx]: Why do we need to set `aria-invalid`?

Comment (jish) [reply src/TextArea/TextArea.tsx]: This article is referenced in the JIRA story: https://hiddedevries.nl/en/blog/2017-04-04-how-to-make-inline-error-messages-accessible

File: src/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -159,6 +159,7 @@ export class TextArea extends React.Component<Props, State> {
   render() {
     let wrapperClass = "TextArea";
     const errorMessage = this.currentErrorMessage();
+    const valid = errorMessage === "";
```
Comment (afumagalli) [inline src/TextArea/TextArea.tsx]: can `errorMessage` be `null`? should we do `const valid = !errorMessage` instead?

File: src/TextArea/TextArea.tsx [hunk 4]
```diff
@@ -221,7 +223,7 @@ export class TextArea extends React.Component<Props, State> {
           <label className="TextArea--label" htmlFor={this.props.name}>
             {this.props.label}
           </label>
-          {inputNote}
+          <span aria-live="polite">{inputNote}</span>
```
Comment (afumagalli) [inline src/TextArea/TextArea.tsx]: polite

File: src/TextInput/TextInput.tsx [hunk 5]
```diff
@@ -163,6 +163,7 @@ export class TextInput extends React.Component<Props, State> {
   render() {
     let wrapperClass = "TextInput";
     const errorMessage = this.currentErrorMessage();
+    const valid = errorMessage === "";
```
Comment (afumagalli) [inline src/TextInput/TextInput.tsx]: ditto

**Comment by jish** [comment]
> "Have you tested with a screen reader to confirm that this change resolves the issue?"

Tested with Voice Over in Safari. It announces the aria instructions after we focus into and out of the text input once because that is how our required field logic kicks in currently... 🤔 I think that is a discussion we should bring up with the product and design teams, as we might want to review the required field behavior to make our text inputs more accessible. 🙂

File: src/TextArea/TextArea.tsx (hunk 4 above)
Comment (prime-time) [reply src/TextArea/TextArea.tsx:226]: 🎩 
🐒 
🚌

**Pull request merged**